### DISCUSSION TITLE
Database & Backup

### DISCUSSION THREAD

**Discussion by G4KCM** [theyosh/TerrariumPI#796] [Ideas]
Hi @theyosh,

TerrariumPI continues to work well here.  Just a couple of thoughts that came to me whilst doing some recent housekeeping…

1) Database size.  It seems to continue growing and I wonder if it would be a good idea to be able to limit the size to a specified number of days history?  I seem to remember some past dialogue regarding database management but can’t immediately find anything so apologies if this has been covered, it would though be useful to be able to do this from the system settings.

2) Presently I manually copy the ‘/data’ folder files to my Google drive,  I was thinking whilst doing this it would be useful to have a backup/restore capability to allow simple backup/restore of critical system files?

As always thanks for the great work!

Cheers 

Clive

**Comment by theyosh** [comment]
Hi,

Database clean up is described here: https://theyosh.github.io/TerrariumPI/faq/cleanup-database/

And I just added a short page about restoring it. https://theyosh.github.io/TerrariumPI/faq/restoring-backup-files/

The actions you do are correct. So keep that in place.

**Reply by G4KCM** [reply]
Thanks!  I should have looked in the FAQ 👍

**Reply by theyosh** [reply]
There was a code bug in the db cleanup. So make sure you do a `git pull` before so you have the correct version.

Also, it is running now for about 3 hours, and needs about 6 hours more it looks like ... :( So this is a slow process, in which TerrariumPI needs to be shutdown. For me not an issue, as the lights are on, and the terrarium can stay this way for about 10 hours..

So, maybe there is a faster way of doing this. Trying to copy the data over to a laptop and run it there. But copying big files also takes time.

**Reply by G4KCM** [reply]
Ok, will make sure I get the latest version. Probably wait now until the weather is a bit warmer before I run it so the heating control won’t be so critical.  There is still 8.5Gb of a 16Gb SD card available.

**Reply by theyosh** [reply]
Ok, then I suggest to wait a bit more.. because I am looking into a way to make the cleanup faster.

**Reply by G4KCM** [reply]
Will wait for sure, no imminent danger. I just noticed the db was growing.  Cheers!

**Reply by theyosh** [reply]
Tried to make a faster way. But that does not work. So it is a time consuming action when you have a lot of data. For the future, try to do it twice a year. Then it should not take so much time each time you run it.

**Reply by G4KCM** [reply]
👍
